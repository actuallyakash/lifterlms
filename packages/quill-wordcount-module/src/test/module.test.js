import wordCountModule, { setOptions } from '../module';

/**
 * Mock Quill class.
 *
 * @since 2.0.0
 *
 * @param {Element} container DOM element container.
 * @return {MockQuill} A mock Quill instance.
 */
function MockQuill( container ) {
	this.container = container;

	this.getText = function() {
		return container.innerHTML;
	};

	this.on = function( action, callback ) {
		callback();
	};

	return this;
}

describe( 'module', () => {
	describe( 'setOptions()', () => {
		const testData = [
			[ 'Use defaults', {}, undefined ],
			[ 'Specify min and max', { min: 5, max: 1000 }, undefined ],
			[ 'Specify partial l10n object', { l10n: { min: 'Min' } }, undefined ],
			[ 'Specify onChange function', { onChange: () => true }, true ],
		];
		test.each( testData )( '%s', ( testName, opts, expectedCallbackResult ) => {
			const merged = setOptions( opts );
			expect( setOptions( opts ) ).toMatchSnapshot();
			expect( merged.onChange() ).toStrictEqual( expectedCallbackResult );
		} );
	} );

	describe( 'module()', () => {
		const testData = [
			[ 'No starting text and default options', '', {} ],
			[ 'Latin characters and specified options', 'Lorem ipsum dolor sit.', { min: 1, max: 1000, l10n: { max: 'Max' } } ],
			[ 'Chinese characters and default options', '我去了一家中餐馆，买了一条面包。', { min: 1, max: 14 } ],
		];
		test.each( testData )( '%s', ( testName, startingText, opts ) => {
			const container = document.createElement( 'div' );

			container.id = 'ql-editor';
			container.innerHTML = startingText;

			document.body.appendChild( container );

			wordCountModule( new MockQuill( container ), opts );
			expect( document.body.innerHTML ).toMatchSnapshot();

			document.body.innerHTML = '';
		} );
	} );
} );
