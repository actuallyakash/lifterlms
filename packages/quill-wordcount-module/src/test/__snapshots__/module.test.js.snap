// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module module() Chinese characters and default options 1`] = `"<div id=\\"ql-editor\\">我去了一家中餐馆，买了一条面包。</div><div class=\\"ql-wordcount ql-toolbar ql-snow\\" style=\\"margin-top: -1px; font-size: 85%;\\"><i class=\\"ql-wordcount-min\\" style=\\"opacity: 0.5; margin-right: 10px;\\">Minimum: 1</i><i class=\\"ql-wordcount-max\\" style=\\"opacity: 0.5; margin-right: 10px;\\">Maximum: 14</i><span class=\\"ql-wordcount-counter\\" style=\\"float: right; color: rgb(255, 146, 43);\\">14 words</span></div>"`;

exports[`module module() Latin characters and specified options 1`] = `"<div id=\\"ql-editor\\">Lorem ipsum dolor sit.</div><div class=\\"ql-wordcount ql-toolbar ql-snow\\" style=\\"margin-top: -1px; font-size: 85%;\\"><i class=\\"ql-wordcount-min\\" style=\\"opacity: 0.5; margin-right: 10px;\\">Minimum: 1</i><i class=\\"ql-wordcount-max\\" style=\\"opacity: 0.5; margin-right: 10px;\\">Max: 1,000</i><span class=\\"ql-wordcount-counter\\" style=\\"float: right;\\">4 words</span></div>"`;

exports[`module module() No starting text and default options 1`] = `"<div id=\\"ql-editor\\"></div><div class=\\"ql-wordcount ql-toolbar ql-snow\\" style=\\"margin-top: -1px; font-size: 85%;\\"><span class=\\"ql-wordcount-counter\\" style=\\"float: right;\\">0 words</span></div>"`;

exports[`module setOptions() Specify min and max 1`] = `
Object {
  "colorError": "#e5554e",
  "colorWarning": "#ff922b",
  "l10n": Object {
    "max": "Maximum",
    "min": "Minimum",
    "plural": "words",
    "singular": "word",
  },
  "max": 1000,
  "min": 5,
  "onChange": [Function],
}
`;

exports[`module setOptions() Specify onChange function 1`] = `
Object {
  "colorError": "#e5554e",
  "colorWarning": "#ff922b",
  "l10n": Object {
    "max": "Maximum",
    "min": "Minimum",
    "plural": "words",
    "singular": "word",
  },
  "max": null,
  "min": null,
  "onChange": [Function],
}
`;

exports[`module setOptions() Specify partial l10n object 1`] = `
Object {
  "colorError": "#e5554e",
  "colorWarning": "#ff922b",
  "l10n": Object {
    "max": "Maximum",
    "min": "Min",
    "plural": "words",
    "singular": "word",
  },
  "max": null,
  "min": null,
  "onChange": [Function],
}
`;

exports[`module setOptions() Use defaults 1`] = `
Object {
  "colorError": "#e5554e",
  "colorWarning": "#ff922b",
  "l10n": Object {
    "max": "Maximum",
    "min": "Minimum",
    "plural": "words",
    "singular": "word",
  },
  "max": null,
  "min": null,
  "onChange": [Function],
}
`;
